src/main/java/com/unstoppabledomains/resolution/DomainResolution.java

```diff
@@ -0,0 +1,119 @@
+package com.unstoppabledomains.resolution;
+
+import com.unstoppabledomains.exceptions.NamingServiceException;
+
+public interface DomainResolution {
+
+    /**
+     * Resolves domain for a specific ticker address
+     *
+     * @param domain - domain name such as "brad.crypto"
+     * @param ticker - coin ticker such as ETH
+     * @return address in hex-string format
+     * @throws NamingServiceException - if tld is not recognized or there record for such currency is not presented
+     */
+    String getAddress(String domain, String ticker) throws NamingServiceException;
+
+    /**
+     * Produces a getNamehash for a specific domain
+     *
+     * @param domain - domain name such as "brad.crypto"
+     *               see https://docs.ens.domains/contract-api-reference/name-processing
+     * @return getNamehash of a domain for a specific NamingService
+     * @throws NamingServiceException - if tld of the domain is not recognized
+     */
+    String getNamehash(String domain) throws NamingServiceException;
```

Comment:
@hivtushok Why switch to `get` prefix for getters?

Reply:
This is to make the code follow Java code conventions: https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html